### PR TITLE
Interconnect type

### PR DESCRIPTION
**Pull request by supreethpy**
### Description
API 800 Support for Interconnect Types
 
### Issues Resolved
NA
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass for go 1.11.x + gofmt checks .
- [x] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [X] New endpoints supported are updated in the endpoints-support.md file.
- [X] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: examples/interconnect_type.tf [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        ov_ifmatch = "*"
+}
+
+data "oneview_interconnect_type" "interconnect_type" {
+	name = "Synergy 10Gb Pass-Thru Module"
+}
+
+output "oneview_interconnect_type_value" {
+	value = "${data.oneview_interconnect_type.interconnect_type.type}"
+//	value = "Null"
```
Comment (AnushaY1916) [inline examples/interconnect_type.tf]: Remove unwanted commented line

File: oneview/data_source_interconnect_type.go [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"interconnect_bay_set": {
+				Type:     schema.TypeInt,
+				Optional: true,
```
Comment (AnushaY1916) [inline oneview/data_source_interconnect_type.go]: For Data Sources, all non-required attributes must be Computed: true, Check other fields as well.

File: oneview/data_source_interconnect_type.go [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+	capabilitiesMap := make([]interface{}, len(interconnectType.InterconnectCapabilities.Capabilities))
+	for i, capabilities := range interconnectType.InterconnectCapabilities.Capabilities {
+		capabilitiesMap[i] = capabilities
+	}
+
+	InterconnectCapabilities = append(InterconnectCapabilities, map[string]interface{}{
+
+		"capabilities": schema.NewSet(schema.HashString, capabilitiesMap),
+		//		"capabilities": interconnectType.InterconnectCapabilities.Capabilities[0],
```
Comment (AnushaY1916) [inline oneview/data_source_interconnect_type.go]: Remove unwanted lines

File: oneview/data_source_interconnect_type.go [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+						"uri": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"port_info": {
+				Optional: true,
+				Type:     schema.TypeList,
```
Comment (AnushaY1916) [inline oneview/data_source_interconnect_type.go]: Can this be replaced with TypeSet? Is order of the port_info required?

File: oneview/data_source_interconnect_type.go (hunk 3 above)
Comment (supreethpy) [reply oneview/data_source_interconnect_type.go]: Done

File: examples/interconnect_type.tf (hunk 1 above)
Comment (supreethpy) [reply examples/interconnect_type.tf]: Done

File: oneview/data_source_interconnect_type.go (hunk 2 above)
Comment (supreethpy) [reply oneview/data_source_interconnect_type.go]: changes done .

File: oneview/data_source_interconnect_type.go (hunk 4 above)
Comment (supreethpy) [reply oneview/data_source_interconnect_type.go]: Done

**Pull request merged**